### DISCUSSION TITLE
v0.3.1 Release Checklist

### DISCUSSION THREAD

**Discussion by vestrel00** [vestrel00/contacts-android#288] [Release Train]
## Target release date

I don't have an exact target date. So, for now, let's just say...

**September**

> Need something sooner than the target date?? Leave a comment on this discussion and let me know what you need! We’ll figure something out 😁 

## Changes to be included in the release

> I'm not using milestones to group issues into releases because I'm using milestones for a different purpose for versions prior to v1.0.0

#### New features

1. #300 
2. #303 
3. #312 
4. #306 
5. #307
6. #308

#### Bug fixes

1. #289
2. #291
3. #296 
4. #298 
5. #302
6. #305
7. #314 

#### Improvements

1. #315 
2. #309

#### Breaking changes

1. #297
2. #304
3. #311
4. #310
5. #313

## Release management steps

Stuff that needs to be done before, during, and after the release.

#### Before publishing release

- [x] Double check that there are no regressions for all relevant API functions!
- [x] Double check that relevant documentation in code, README, DEV_NOTES, and all howto pages are correct.
- [x] Ensure that JitPack SNAPSHOT of the main branch is green and installable.

#### Publishing the release

- [x] Create the discussion page for the next release (0.3.2)
- [x] Update the README with installation instructions for 0.3.1 and update links to the current (0.3.1) and next release (0.3.2).
- [x] Go through all markdown files and replace any links to the 0.3.1 discussion page with links to the 0.3.1 release tag.
- [x] Create the 0.3.1 tag.
- [x] Publish the 0.3.1 release.

#### After publishing release

- [x] Update `gh-pages` branch.
- [x] Sanity check install 0.3.1 from JitPack in a new project. _(The first JitPack release build typically fails to install. Don't panic. Just delete the build and rebuild it.)_
- [x] Leave a comment in this discussion with a link to the release.
- [x] Leave a comment in each of the above issues with a link to the release.
- [x] Unpin this discussion and pin the next release's discussion page.
- [x] Close the [previous release's discussion / checklist](https://github.com/vestrel00/contacts-android/discussions/218)

**Comment by vestrel00** [comment]
These discussions contains some samples of the new features that are included in this release;

- https://github.com/vestrel00/contacts-android/discussions/299#discussioncomment-6867208
- https://github.com/vestrel00/contacts-android/discussions/301#discussioncomment-6952055

**Comment by vestrel00** [comment]
The release is out now! https://github.com/vestrel00/contacts-android/releases/tag/0.3.1

**Comment by 2zt** [comment]
Thanks so much for your meticulous work. I've been following the evolution to this version and I'm impressed by the precision and completeness of each issue. You are especially methodical. I plan to use the library in a project later this month, and I needed exactly the features of this release for my synchronization needs. Thanks for this amazing library

**Reply by vestrel00** [reply]
@2zt, wow, thank you so much! Comments like yours are one of main reasons that I work on this library in my spare time! I'm super glad that my work can be of some use to you.

My goal is to give back to the developer community that lifted me and my family out of extreme poverty. It makes me very fulfilled to know that I'm able to help other people 😁 

Anyways, if you encounter any issues, just start a discussion or create an issue 😉
